miden-lib/asm/sat/account.masm

```diff
@@ -265,3 +293,32 @@ export.set_item
     swapw exec.layout::set_acct_storage_root
     # => [V]
 end
+
+#! Authenticates the proedcure root is part of the account code Merkle treee. Panics if the
+#! procedure root is not part of the account code Merkle tree.
+#!
+#! Stack: [PROC_ROOT]
+#! Output: [PROC_ROOT]
+#!
+#! - PROC_ROOT is the hash of the procedure to authenticate.
+export.authenticate_procedure.1
+    # load the account code root onto the stack
+    exec.layout::get_acct_code_root swapw
+    # => [PROC_ROOT, CODE_ROOT]
+
+    # load the index of the procedure root onto the advice stack
+    adv.keyval adv_push.1 movdn.4
+    # => [PROC_ROOT, index, CODE_ROOT]
```

Comment:
Thought for the future: I wonder if there is a better way to look up index for a given leaf (e.g., maybe via a dedicated decorator). This way, we won't have to pre-load procedure too to index mappings into the advice map.